app/src/main/scala/org/alephium/explorer/util/Scheduler.scala

```diff
@@ -0,0 +1,187 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.util
+
+import java.time.{LocalDate, LocalDateTime, LocalTime}
+import java.util.{Timer, TimerTask}
+
+import scala.annotation.tailrec
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+import com.typesafe.scalalogging.StrictLogging
+
+import org.alephium.explorer.util.Scheduler.scheduleTime
+
+object Scheduler extends StrictLogging {
+
+  /**
+    * Creates a Scheduler.
+    *
+    * @param name     the name of the associated thread
+    * @param isDaemon true if the associated thread should run as a daemon
+    */
+  @SuppressWarnings(Array("org.wartremover.warts.DefaultArguments"))
+  def apply(name: String, isDaemon: Boolean = true): Scheduler =
+    new Scheduler(
+      name       = name,
+      timer      = new Timer(name, isDaemon),
+      terminated = false
+    )
+
+  /**
+    * Calculate time left to schedule for the input [[LocalTime]].
+    * If the time is in the past then the schedule for tomorrow.
+    *
+    * TimeZone used is local machine's default time-zone
```

Comment:
mmm I think we should try to work with UTC time only, for example for the `scheduleDailyAt` I will set it up at 1am, because I really want it to run just when a UTC day is finished. 

As I don't know what's the time-zone of the AWS machine for prod it's impossible :/

Reply:
Ah yes good idea. For this use-case made the following changes

- Added `scheduleDailyAtUTC(utcTime)`.
- Changed the function to use `LocalDateTime` instead of `LocalTime` to specify the **day** to start scheduling.

WDYT?
 
The highlighted function `scheduleTime` calculates `FiniteDuration` left until next schedule right, so here UTC or Local wouldn't make a difference to the final output. Prefer using local system time here because it's easier when debugging & testing, easy to visually compare time against the time on my machine.

Reply:
As requested on Slack I've switched
- `LocalTime` to `OffsetTime`
- `LocalDateTime` to `ZonedDateTime`